### PR TITLE
Swapped out empty interfaces for generics and added go.mod

### PR DESCRIPTION
**Pull request by skulos**
Replaced all of the `interface{}` types with a `comparable` type.

### CODE REVIEW

File: bimap.go [hunk 1]
```diff
@@ -4,20 +4,29 @@ package bimap
...
 
 // NewBiMap returns a an empty, mutable, biMap
-func NewBiMap() *BiMap {
-	return &BiMap{forward: make(map[interface{}]interface{}), inverse: make(map[interface{}]interface{}), immutable: false}
+func NewBiMap[K comparable, V comparable]() *BiMap[K, V] {
+	return &BiMap[K, V]{forward: make(map[K]V), inverse: make(map[V]K), immutable: false}
+}
+
+// NewBiMapFrom returns a new BiMap from a map[K, V]
+func NewBiMapFromMap[K comparable, V comparable](forwardMap map[K]V) *BiMap[K, V] {
```
Comment (vishalkuo) [inline bimap.go:20]: could you rebase? I recently merged https://github.com/vishalkuo/bimap/pull/4 which does something similar. Let's just edit that function

**Review by vishalkuo — APPROVED** [review]
awesome, thanks

**Review by vishalkuo — CHANGES_REQUESTED** [review]
Sorry accidentally approved, meant to request changes. LGTM overall just needs a small change

File: bimap.go (hunk 1 above)
Comment (skulos) [reply bimap.go:20]: No problem, Will do it now.

**Comment by skulos** [comment]
@vishalkuo I accidentally deleted your new functions with merge (#4). It comes down to the same. I could just redo it if you prefer.

**Review by vishalkuo — APPROVED** [review]

**Pull request merged**